File: src/main/java/info/dddeurope/lab/app/commandsHandlers/RegisterUserCommandHandler.java
package info.dddeurope.lab.app.commandsHandlers;

import info.dddeurope.lab.app.commands.RegisterUserCommand;
import info.dddeurope.lab.app.domain.UserAggregate.User;
import info.dddeurope.lab.app.dtos.UserDto;
import info.dddeurope.lab.app.events.UserRegisteredEvent;
import info.dddeurope.lab.app.repositories.UserRepository;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class RegisterUserCommandHandler {

    UserRepository userRepository;

    public UserRegisteredEvent handle(RegisterUserCommand registerUserCommand) throws Exception {
        
        final String userId = registerUserCommand.getUserId();
        final UserDto userDto = registerUserCommand.getUserDto();

        User user = new User(userId,
            userDto.getFirstName(),
            userDto.getLastName(),
            userDto.getEmail(),
            userDto.getDateOfBirth(),
            userDto.getAddress());

        this.userRepository.set(userId, user);
        return new UserRegisteredEvent(userId, userDto); 
        
    }   

}

File: DotNet/App/CommandsHandlers/RegisterUserCommandHandler.cs
using App.Commands;
using App.Domain.UserAggregate;
using App.Events;
using App.Repositories;

namespace App.CommandsHandlers
{
    public class RegisterUserCommandHandler
    {
        readonly UserRepository _userRepository;

        public RegisterUserCommandHandler(UserRepository userRepository)
        {
            _userRepository = userRepository;
        }

        public UserRegisteredEvent Handle(RegisterUserCommand registerUserCommand)
        {
            var userId = registerUserCommand.UserId;
            var userDto = registerUserCommand.UserDto;

            var user = new User(userId,
                userDto.FirstName,
                userDto.LastName,
                userDto.Email,
                userDto.DateOfBirth,
                userDto.Address);

            _userRepository.Set(userId, user);
            return new UserRegisteredEvent(userId, userDto);
        }
    }
}